addons/mail/models/ir_actions_server.py

```diff
@@ -420,6 +420,8 @@ def _run_action_next_activity(self, eval_context=None):
                 # if x2m field, assign to the first user found
                 # (same behavior as Field.traverse_related)
                 vals['user_id'] = user.ids[0]
+            else:
+                vals['user_id'] = self.env.user.id
```

Comment:
Well we definitely don't want this.

The sales-person confirming the sale shouldn't be assigned a delivery activity for this particular example.